server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CachedBlobStore.java

```diff
@@ -154,12 +156,12 @@ public InputStream read(BucketName bucketName, BlobId blobId) throws ObjectStore
         return Mono.just(bucketName)
             .filter(getDefaultBucketName()::equals)
             .flatMap(deleteBucket -> readBytesInDefaultBucket(bucketName, blobId))
-            .switchIfEmpty(readBytesFromBackend(bucketName, blobId));
+            .switchIfEmpty(Mono.defer(() -> readBytesFromBackend(bucketName, blobId)));
```

Comment:
That is suspicious

Reply:
I cant find any thing official than: 
```
SwitchIfEmpty accepts Mono "by value". Meaning that even before you subscribe to your mono, this alternative mono's evaluation is already triggered. 
```
So it run in to a bug:
- We still readBytesFromBackend when we readFromCache, it make latency metrics for backend recorded

Reply:
Well, after tested with you, it proved my conclusion it wrong, we dont needed it.